staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java

```diff
@@ -109,6 +126,68 @@ public Path getTenantInprogressDirectory(final TenantId tenantId)
         return getPathForTenant(tenantId).resolve(INPROGRESS_FOLDER);
     }
 
+    public void monitorBatchProgressInFileSystem(final TenantId tenantId, final BatchId batchId, String threadID)
```

Comment:
All parameters should be `final`:

```suggestion
    public void monitorBatchProgressInFileSystem(final TenantId tenantId, final BatchId batchId, final String threadID)
```

Reply:
Done.